### ISSUE TITLE
Subscription count metrics is always returning 0

### ISSUE THREAD

**Issue by jayeshkm** [halfgaar/FlashMQ#125]
Subscribing to topic ```$SYS/broker/subscriptions/count``` always returns subscription count as 0 even though there are valid subscriptions

**Comment by halfgaar** [comment]
You're right, it should be fixed. I had to disable it at some point because counting it accurately was causing too much load. At least with that implementation. It needs refactor.

**Comment by jayeshkm** [comment]
I have not went in depth into the code, but I see ``subscriptionCount`` variable in subscription store. Can we just return that?

**Comment by halfgaar** [comment]
No, it's more complicated. It has to do with how sessions are added to the subscription tree as weak pointers. They will  expire sooner or later, but this can't easily be tracked.

I may make the subscription counter indicative instead of 100% correct. It can be recalculated upon certain internal maintenance jobs, but that means that the count will be (a bit) higher than factually correct.

**Comment by halfgaar** [comment]
I fixed it in master. I chose an implementation that has no impact on performance. That means it's 100% correct on all subscription mutations, but it's corrected every 30-ish minutes.

I applied the same logic to the subscription counter.

It will appear in the next release.

**Comment by halfgaar** [comment]
Fixed in release 1.19.0.
